src/crawlee/beautifulsoup_crawler/beautifulsoup_crawler.py

```diff
@@ -129,8 +130,12 @@ async def enqueue_links(
                 if label is not None:
                     link_user_data.setdefault('label', label)
 
-                if (href := link.attrs.get('href')) is not None:
-                    requests.append(BaseRequestData.from_url(href.strip(), user_data=link_user_data))
+                if (url := link.attrs.get('href')) is not None:
+                    if not is_url_absolute(url):
+                        url = str(make_url_absolute(context.request.url, url))
+
+                    url = url.strip()
```

Comment:
Could we do this before the absolutifying?

Reply:
yep, it makes sense